### PR TITLE
Insertable Streams for MediaStreamTrack API

### PR DESCRIPTION
**Pull request by rachelandrew**
Adds the API overview and two interfaces.

Reviewer: @jpmedley

### CODE REVIEW

File: files/en-us/web/api/mediastreamtrackgenerator/index.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  - Interface
+  - Reference
+  - MediaStreamTrackGenerator
+browser-compat: api.MediaStreamTrackGenerator
+---
+{{DefaultAPISidebar("Insertable Streams for MediaStreamTrack API")}}
+
+The **`MediaStreamTrackGenerator`** interface of the {{domxref('Insertable Streams for MediaStreamTrack API')}} creates a {{domxref("WritableStream")}} that acts as a {{domxref("MediaStreamTrack")}} source.
+
+The object consumes a stream of media frames as input, which can be audio or video frames.
```
Comment (jpmedley) [inline files/en-us/web/api/mediastreamtrackgenerator/index.md:15]:
```suggestion
The object consumes a stream of media frames as input, which can be audio or video frames.
```

File: files/en-us/web/api/mediastreamtrackgenerator/index.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+- {{domxref("MediaStreamTrackGenerator.writable")}}
+  - : A {{domxref("WritableStream")}}.
+
+## Methods
+
+_This interface doesn't implement any specific methods, but inherits methods from {{domxref("MediaStreamTrack")}}._
+
+## Examples
+
+The following example is from the article [Insertable streams for MediaStreamTrack](https://web.dev/mediastreamtrack-insertable-media-processing/), and demonstrates a barcode scanner application, which identifies and processes barcodes before writing the transformed frames to the writablestream of {{domxref("MediaStreamTrackGenerator.writable")}}.
```
Comment (jpmedley) [inline files/en-us/web/api/mediastreamtrackgenerator/index.md:35]:
```suggestion
The following example is from the article [Insertable streams for MediaStreamTrack](https://web.dev/mediastreamtrack-insertable-media-processing/), and demonstrates a barcode scanner application, which process barcodes and  highlights them before writing the transformed frames to the writablestream of {{domxref("MediaStreamTrackGenerator.writable")}}.
```

File: files/en-us/web/api/mediastreamtrackgenerator/mediastreamtrackgenerator/index.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+browser-compat: api.MediaStreamTrackGenerator.MediaStreamTrackGenerator
+---
+{{DefaultAPISidebar("Insertable Streams for MediaStreamTrack API")}}
+
+The **`MediaStreamTrackGenerator()`** constructor creates a new {{domxref("MediaStreamTrackGenerator")}} object which consumes a stream of media frames and exposes a {{domxref("MediaStreamTrack")}}.
+
+## Syntax
+
+```js
+new MediaStreamTrackGenerator(init);
```
Comment (jpmedley) [inline files/en-us/web/api/mediastreamtrackgenerator/mediastreamtrackgenerator/index.md:18]:
```suggestion
new MediaStreamTrackGenerator(options);
```

Comment (jpmedley) [reply files/en-us/web/api/mediastreamtrackgenerator/mediastreamtrackgenerator/index.md:18]: "options" outnumbers "init" by about 4:1.

File: files/en-us/web/api/mediastreamtrackgenerator/mediastreamtrackgenerator/index.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+### Parameters
+
+- `init`
```
Comment (jpmedley) [inline files/en-us/web/api/mediastreamtrackgenerator/mediastreamtrackgenerator/index.md:23]:
```suggestion
- `options`
```

File: files/en-us/web/api/mediastreamtrackgenerator/mediastreamtrackgenerator/index.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  - : An object containing the property `kind`, which is a {{domxref("DOMString", "string")}} with one of the following values:
+      - `"audio"`
+        - : The stream accepts {{domxref("AudioTrack")}} objects.
```
Comment (jpmedley) [inline files/en-us/web/api/mediastreamtrackgenerator/mediastreamtrackgenerator/index.md:26]:
```suggestion
        - : Specifies that the stream accepts {{domxref("AudioTrack")}} objects.
```

File: files/en-us/web/api/mediastreamtrackgenerator/mediastreamtrackgenerator/index.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      - `"video"`
+        - : The stream accepts {{domxref("VideoTrack")}} objects.
```
Comment (jpmedley) [inline files/en-us/web/api/mediastreamtrackgenerator/mediastreamtrackgenerator/index.md:28]:
```suggestion
        - : Specifies that the stream accepts {{domxref("VideoTrack")}} objects.
```

File: files/en-us/web/api/mediastreamtrackgenerator/writable/index.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  - API
+  - Property
+  - Reference
+  - writable
+  - MediaStreamTrackGenerator
+browser-compat: api.MediaStreamTrackGenerator.writable
+---
+{{DefaultAPISidebar("Insertable Streams for MediaStreamTrack API")}}
+
+The **`writable`**  property of the {{domxref("MediaStreamTrackGenerator")}} interface returns a {{domxref("WritableStream")}}. This allows the writing of media frames to the `MediaStreamTrackGenerator`. The frames will be audio or video, the type is dictated by the kind of `MediaStreamTrackGenerator` that has been created.
```
Comment (jpmedley) [inline files/en-us/web/api/mediastreamtrackgenerator/writable/index.md:14]:
```suggestion
The **`writable`**  property of the {{domxref("MediaStreamTrackGenerator")}} interface returns a {{domxref("WritableStream")}}. This allows the writing of media frames to the `MediaStreamTrackGenerator`. The frames will be audio or video. The type is dictated by the kind of `MediaStreamTrackGenerator` that was created.
```

File: files/en-us/web/api/mediastreamtrackprocessor/index.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+## Constructor
+
+- {{domxref("MediaStreamTrackProcessor.MediaStreamTrackProcessor()")}}
+  - : Creates a new `MediaStreamTrackProcessor` object.
+
+## Properties
+
+- {{domxref("MediaStreamTrackProcessor.readable")}}
+  - : A {{domxref("ReadableStream")}}.
```
Comment (jpmedley) [inline files/en-us/web/api/mediastreamtrackprocessor/index.md:23]:
```suggestion
  - : Returns a {{domxref("ReadableStream")}}.
```

File: files/en-us/web/api/mediastreamtrackprocessor/index.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+## Examples
+
+The following example is from the article [Insertable streams for MediaStreamTrack](https://web.dev/mediastreamtrack-insertable-media-processing/), and demonstrates a barcode scanner application, which transforms the stream accessed via {{domxref("MediaStreamTrackProcessor.readable")}}.
```
Comment (jpmedley) [inline files/en-us/web/api/mediastreamtrackprocessor/index.md:27]:
```suggestion
The following example is from the article [Insertable streams for MediaStreamTrack](https://web.dev/mediastreamtrack-insertable-media-processing/), and demonstrates a barcode scanner application, which transforms the stream accessed via {{domxref("MediaStreamTrackProcessor.readable")}} by highlighting the barcode.
```

File: files/en-us/web/api/mediastreamtrackprocessor/mediastreamtrackprocessor/index.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+tags:
+  - API
+  - Constructor
+  - Reference
+  - MediaStreamTrackProcessor
+browser-compat: api.MediaStreamTrackProcessor.MediaStreamTrackProcessor
+---
+{{DefaultAPISidebar("Insertable Streams for MediaStreamTrack API")}}
+
+The **`MediaStreamTrackProcessor()`** constructor creates a new {{domxref("MediaStreamTrackProcessor")}} object which
```
Comment (jpmedley) [inline files/en-us/web/api/mediastreamtrackprocessor/mediastreamtrackprocessor/index.md:13]:
```suggestion
The **`MediaStreamTrackProcessor()`** constructor creates a new {{domxref("MediaStreamTrackProcessor")}} object which consumes a {{domxref("MediaStreamTrack")}} object's source and generates a stream of media frames.
```

File: files/en-us/web/api/mediastreamtrackprocessor/mediastreamtrackprocessor/index.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+```js
+new MediaStreamTrackProcessor(init);
+```
+
+### Parameters
+
+- `track`
+  - : A {{domxref("MediaStreamTrack")}}.
+- `maxBufferSize`{{Optional_Inline}}
+  - : An integer specifying the maximum number of media frames to be buffered.
```
Comment (jpmedley) [inline files/en-us/web/api/mediastreamtrackprocessor/mediastreamtrackprocessor/index.md:26]:
```suggestion
- `options`
  - : An object with the following properties:
    - `track`
      - : A {{domxref("MediaStreamTrack")}}.
    - `maxBufferSize`{{Optional_Inline}}
      - : An integer specifying the maximum number of media frames to be buffered.
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/insertable_streams_for_mediastreamtrack_api/index.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
+---
+title: Insertable Streams for MediaStreamTrack API
+slug: Web/API/Insertable_Streams_for_MediaStreamTrack_API
+tags:
+  - API
+  -
+  - Overview
+  - Reference
+browser-compat: api.ContactAddress
```
[the remark this answers is not in the document]
Comment (rachelandrew) [reply files/en-us/web/api/insertable_streams_for_mediastreamtrack_api/index.md:9]:
```suggestion
```

**Review by jpmedley — APPROVED** [review]

**Pull request merged**